src/main/java/org/skriptlang/skript/misc/colors/ColorUtils.java

```diff
@@ -0,0 +1,384 @@
+		float b = color.getBlue() / 255f;
+
+		// max defines brightness, delta defines saturation
+		float max = Math.max(r, Math.max(g, b));
+		float min = Math.min(r, Math.min(g, b));
+		float delta = max - min;
+		float hue = 0f;
+		float saturation = max == 0f ? 0f : delta / max;
+
+		// hue depends on which channel is max
+		if (delta != 0f) {
+			if (max == r) {
+				hue = ((g - b) / delta) % 6f;
+			} else if (max == g) {
+				hue = ((b - r) / delta) + 2f;
+			} else {
+				hue = ((r - g) / delta) + 4f;
+			}
+			hue *= 60f;
+			if (hue < 0f) hue += 360f;
+		}
+		hue /= 360f;
+		return new float[]{ hue, saturation, max};
+	}
+
+	/**
+	 * Converts HSB values to a {@link ColorRGB} object.
+	 * This is different to {@link #hslToRgb(float[])}.
+	 *
+	 * @param hsb a float array representing HSB values
+	 * @return a {@link ColorRGB} object given the HSB values
+	 */
+	private static @NotNull ColorRGB hsbToRgb(float @NotNull [] hsb) {
+		float hue = hsb[0], saturation = hsb[1], brightness = hsb[2];
+		// determine hue sector and interpolate
+		float red = 0f, green = 0f, blue = 0f;
+		int hueSector = (int) (hue * 6f);
+		float sectorFraction = hue * 6f - hueSector;
+		float lowerBound = brightness * (1f - saturation);
+		float higherBound = brightness * (1f - sectorFraction * saturation);
+		float hueOffset = brightness * (1f - (1f - sectorFraction) * saturation);
+
+		// assign rgb values based on sector
+		switch (hueSector % 6) {
+			case 0:
+				red = brightness; green = hueOffset; blue = lowerBound; break;
+			case 1:
+				red = higherBound; green = brightness; blue = lowerBound; break;
+			case 2:
+				red = lowerBound; green = brightness; blue = hueOffset; break;
+			case 3:
+				red = lowerBound; green = higherBound; blue = brightness; break;
+			case 4:
+				red = hueOffset; green = lowerBound; blue = brightness; break;
+			case 5:
+				red = brightness; green = lowerBound; blue = higherBound; break;
+		}
+		int r = Math.round(red * 255f);
+		int g = Math.round(green * 255f);
+		int b = Math.round(blue * 255f);
+		return ColorRGB.fromRGBA(r, g, b, 255);
+	}
+
+	/**
+	 * Converts a {@link Color} to its grayscale equivalent.
+	 *
+	 * @param color the {@link Color} to convert to grayscale
+	 * @return the colour's grayscale equivalent
+	 */
+	public static @NotNull ColorRGB toGrayscale(@NotNull Color color) {
+		// weighted average simulates human perception
+		int gray = (int)(0.299 * color.getRed() + 0.587 * color.getGreen() + 0.114 * color.getBlue());
+		return ColorRGB.fromRGBA(gray, gray, gray, color.getAlpha());
+	}
+
+	/**
+	 * Converts a {@link Color} to its sepiatone equivalent.
+	 *
+	 * @param color the {@link Color} to convert to sepiatone
+	 * @return the colour's sepiatone equivalent
+	 */
+	public static @NotNull ColorRGB toSepia(@NotNull Color color) {
+		// standard sepia formula
+		int r = color.getRed();
+		int g = color.getGreen();
+		int b = color.getBlue();
+		int sepiaRed = (int) (0.393 * r + 0.769 * g + 0.189 * b);
```

Comment:
Is there a source for these values that might be worth linking to in a comment (even just like Wikipedia)?